lib/SmartyPlugins/LMSSmartyPlugins.php

```diff
@@ -1152,6 +1152,38 @@ public static function networkDeviceTypesFunction(array $params, $template)
             . '</select>';
     }
 
+    public static function networkDeviceSelectionFunction(array $params, $template)
+    {
+        $LMS = LMS::getInstance();
+        $netdevicelist = $LMS->GetNetDevList();
+        unset($netdevicelist['total']);
```

Comment:
Przy wielokrotnym użyciu tej funkcji smarty ciągle będzie wywoływać dość kosztowną funkcję `GetNetDevList()`.
W innych miejscach mamy już na to sztuczkę.

Reply:
Wskażesz mi przykład jak chcesz by to było zrobione? Przeglądałem wystąpienia `$LMS` i `$lms` i nie widzę antidotum na wielokrotne odpalanie tej funkcji. Zastosuje ten sam sposób zamiast wymyślać koło od 0.

Reply:
hint: szukaj w LMSSmartyPlugins słowa `static`.

Reply:
4a2d3b56cfc51d24edbefd19b6012d08e156593f

Może być i tak myślałem, że o jakiś cache chodzi.